packages/cli/lib/services/response-saver.service.ts

```diff
@@ -60,35 +57,98 @@ export function onAxiosRequestFulfilled({
-            concatenateIfExists: false
+            customFilePath: 'mocks/nango/getMetadata.json'
         });
 
         return response;
     }
 
-    const [pathname, params] = response.request.path.split('?');
+    const [pathname] = response.request.path.split('?');
     const strippedPath = pathname.replace('/', '');
 
-    let concatenateIfExists = false;
-
-    if (params && params.includes('page')) {
-        concatenateIfExists = true;
-    }
+    const requestIdentity = computeConfigIdentity(response.config);
 
-    saveResponse<AxiosResponse>({
+    saveResponse<CachedRequest>({
         directoryName,
-        data: response.data,
-        customFilePath: `mocks/nango/${method}/${strippedPath}/${syncName}.json`,
-        concatenateIfExists
+        data: {
+            ...requestIdentity,
+            response: response.data
+        },
+        customFilePath: `mocks/nango/${method}/${strippedPath}/${syncName}/${requestIdentity.requestIdentityHash}.json`
     });
 
     return response;
 }
+
+function computeConfigIdentity(config: AxiosRequestConfig): ConfigIdentity {
+    const method = config.method?.toLowerCase() || 'get';
+    const params = sortEntries(Object.entries(config.params || {}));
+
+    const requestIdentity: [string, unknown][] = [
+        ['method', method],
+        ['url', config.url],
+        ['params', params]
+    ];
+
+    const dataIdentity = computeDataIdentity(config);
+    if (dataIdentity) {
+        requestIdentity.push(['data', dataIdentity]);
+    }
+
+    if (config.headers !== undefined) {
+        const filteredHeaders = Object.entries(config.headers).filter(([key]) => !FILTER_HEADERS.includes(key.toLowerCase()));
+        sortEntries(filteredHeaders);
+        requestIdentity.push([`headers`, filteredHeaders]);
+    }
+
+    // sort by key so we have a consistent hash
+    sortEntries(requestIdentity);
+
+    const requestIdentityHash = crypto.createHash('sha1').update(JSON.stringify(requestIdentity)).digest('hex');
+
+    return {
+        requestIdentityHash,
+        requestIdentity
+    };
+}
+
+function sortEntries(entries: [string, unknown][]): [string, unknown][] {
+    return entries.sort((a, b) => (a[0] < b[0] ? -1 : a[0] > b[0] ? 1 : 0));
+}
+
+function computeDataIdentity(config: AxiosRequestConfig): string | undefined {
+    const data = config.data;
+
+    if (!data) {
+        return undefined;
+    }
+
+    let dataString = '';
+    if (typeof data === 'string') {
+        dataString = data;
+    } else if (Buffer.isBuffer(data)) {
+        dataString = data.toString('base64');
+    } else {
+        try {
+            dataString = JSON.stringify(data);
+        } catch (e) {
+            if (e instanceof Error) {
+                throw new Error(`Unable to compute request identity: ${e.message}`);
+            } else {
+                throw new Error('Unable to compute request identity');
+            }
+        }
+    }
+
+    if (dataString.length > 1000) {
```

Comment:
why not always hashing?

Reply:
It's easier to spot why things changed in the recorded payloads if you dump the literal payload, rather than just a hash, but when the payloads get extremely large it's not really reasonable to store them anymore.